modules/barcode/src/detector/bardetect.cpp

```diff
@@ -0,0 +1,422 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+// Copyright (c) 2020-2021 darkliang wangberlinT Certseeds
+
+#include "../precomp.hpp"
+#include "bardetect.hpp"
+
+
+namespace cv {
+namespace barcode {
+static constexpr float PI = static_cast<float>(CV_PI);
+static constexpr float HALF_PI = static_cast<float>(CV_PI / 2);
+
+#define CALCULATE_SUM(ptr, result) \
+    ptr += left_col + integral_cols * top_row;\
+    top_left = static_cast<float>(*(ptr));\
+    (ptr) += right_col - left_col;\
+    top_right = static_cast<float>(*(ptr));\
+    (ptr) += (bottom_row - top_row) * integral_cols;\
+    bottom_right = static_cast<float>(*(ptr));\
+    (ptr) -= right_col - left_col;\
+    bottom_left = static_cast<float>(*(ptr));\
+    (ptr) -= (bottom_row - top_row) * integral_cols;\
+    (ptr) -= left_col + integral_cols * top_row;\
+    (result) = (bottom_right - bottom_left - top_right + top_left);
+
+
+inline bool Detect::isValidCoord(const Point &coord, const Size &limit)
+{
+    if ((coord.x < 0) || (coord.y < 0))
+    {
+        return false;
+    }
+
+    if ((unsigned) coord.x > (unsigned) (limit.width - 1) || ((unsigned) coord.y > (unsigned) (limit.height - 1)))
+    {
+        return false;
+    }
+
+    return true;
+}
+
+inline float Detect::computeOrientation(float y, float x)
+{
+    if (x >= 0)
+    {
+        return atan(y / x);
+    }
+    if (y >= 0)
+    {
+        return (atan(y / x) + PI);
+    }
+    return (atan(y / x) - PI);
```

Comment:
why is not [atan2](http://www.cplusplus.com/reference/cmath/atan2/)?

Also there is `cv::fastAtan()` (but result is in degrees instead of radians and probably modified/shifted range)